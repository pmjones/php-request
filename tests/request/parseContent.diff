001+ string(0) ""
003+ string(0) ""
005+ string(1) "/"
006- NULL
007- NULL
008- NULL
009- NULL
010- NULL
007+ string(0) ""
009+ string(5) "text/"