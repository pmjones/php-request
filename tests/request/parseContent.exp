NULL
NULL
NULL
NULL
NULL
NULL
NULL
NULL
NULL
NULL
string(10) "text/plain"
NULL
string(10) "text/plain"
string(5) "utf-8"
string(10) "text/plain"
NULL
string(10) "text/plain"
string(5) "utf-8"
string(12) "image/x-tiff"
NULL