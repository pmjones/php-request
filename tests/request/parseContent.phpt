--TEST--
SapiRequest::parseContentType
--FILE--
<?php
$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => null]]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => '']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => '/']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => ';']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => 'text/']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => 'text/plain']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => 'text/plain;charset=utf-8']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => 'text/plain;foo=bar']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => 'text/plain;foo=bar;charset=utf-8;baz=dib']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

$request = new SapiRequest(['_SERVER' => ['HTTP_CONTENT_TYPE' => 'image/x-tiff;baz = "bat"']]);
var_dump($request->contentType);
var_dump($request->contentCharset);

--EXPECT--
NULL
NULL
NULL
NULL
NULL
NULL
NULL
NULL
NULL
NULL
string(10) "text/plain"
NULL
string(10) "text/plain"
string(5) "utf-8"
string(10) "text/plain"
NULL
string(10) "text/plain"
string(5) "utf-8"
string(12) "image/x-tiff"
NULL
