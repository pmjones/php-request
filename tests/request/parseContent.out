string(0) ""
NULL
string(0) ""
NULL
string(1) "/"
NULL
string(0) ""
NULL
string(5) "text/"
NULL
string(10) "text/plain"
NULL
string(10) "text/plain"
string(5) "utf-8"
string(10) "text/plain"
NULL
string(10) "text/plain"
string(5) "utf-8"
string(12) "image/x-tiff"
NULL